test/basic.html

```diff
@@ -9,106 +9,156 @@
+      });
+
+      suite('disable auto-close', function() {
+        var spy;
+        setup(function() {
           toast = fixture('basic');
+          spy = sinon.spy(toast, 'async');
+        });
+        test('duration = Infinity', function() {
+          toast.duration = Infinity;
           toast.show();
-          assert.isTrue(toast.opened, '`opened` is true');
+          assert.isFalse(spy.calledWith(toast.close), '`async` was not called with `close()`');
+          assert.isFalse(spy.calledWith(toast.hide), '`async` was not called with `hide()`');
         });
 
-        test('show() will auto-close toast after `duration` milliseconds', function(done) {
-          toast = fixture('basic');
-          toast.duration = 10;
-          var debounceSpy = sinon.spy(toast, 'debounce');
+        test('duration = 0', function() {
+          toast.duration = 0;
           toast.show();
-          assert.isTrue(debounceSpy.called, '`debounce` was called');
-          setTimeout(function(){
-            assert.isFalse(toast.opened, '`opened` is false');
-            done();
-          }, 12);
+          assert.isFalse(spy.calledWith(toast.close), '`async` was not called with `close()`');
+          assert.isFalse(spy.calledWith(toast.hide), '`async` was not called with `hide()`');
         });
 
-        suite('disable auto-close', function(){
-          var spy;
-          setup(function(){
-            toast = fixture('basic');
-            spy = sinon.spy(toast, 'debounce');
-          });
-          test('duration = Infinity', function() {
-            toast.duration = Infinity;
-            toast.show();
-            assert.isFalse(spy.called, '`debounce` was not called');
-          });
-
-          test('duration = 0', function() {
-            toast.duration = 0;
-            toast.show();
-            assert.isFalse(spy.called, '`debounce` was not called');
-          });
-
-          test('duration = -10', function() {
-            toast.duration = -10;
-            toast.show();
-            assert.isFalse(spy.called, '`debounce` was not called');
-          });
+        test('duration = -10', function() {
+          toast.duration = -10;
+          toast.show();
+          assert.isFalse(spy.calledWith(toast.close), '`async` was not called with `close()`');
+          assert.isFalse(spy.calledWith(toast.hide), '`async` was not called with `hide()`');
         });
+      });
+
+      test('there is only 1 toast opened', function() {
+        var toast1 = fixture('basic');
+        var toast2 = fixture('show');
+        toast2.open();
+        toast1.open();
+        assert.isTrue(toast1.opened, 'toast1 is opened');
+        assert.isFalse(toast2.opened, 'toast2 is not opened');
+        toast2.open();
+        assert.isFalse(toast1.opened, 'toast1 is now not opened');
+        assert.isTrue(toast2.opened, 'toast2 is now opened');
+      });
+
+      test('toast gets closed only once when doing show/hide', function(done) {
```

Comment:
this test is valuable with the next one - probably the should be in the same suite.
The final goal is to keep the toast opened if the last action performed was `show()` or `open()`. I've separated the test in 2 checks:
- check that close is invoked once
- check that the toast is opened after a series of open/close/open

WDYT? too many checks?